### DISCUSSION TITLE
FATES initialization with inventory

### DISCUSSION THREAD

**Discussion by LianGong98** [NGEET/fates#1120] [Q&A] [answered]
Dear Developers,
  When I  initialized FATES using the PSS and CSS files I created.During execution, I received the error message:

 _total area is wrong in update_hlm_dynamics  3.197600000000000E-003
 ENDRUN:
 ERROR in EDCanopyStructureMod.F90 at line 1971_ 

  I suspect there might be an issue with how I've calculated the area in my PSS file. I've utilized a calculation of 1 divided by the total number of patches (e.g., if patches are labeled as patch01, patch01, patch02, patch02, patch03, and patch04, then the area calculation should yield area = 1/4). Furthermore, regarding the CSS file, does it exclusively include living trees or all trees? Also, it seems that merging identical cohorts might not be necessary, which means each tree correspond to its own cohort?
Any assistance would be greatly appreciated！

**Comment by mpaiao** [comment, marked as answer]
@LianGong98 I had this issue before, and the culprit happened to be truncation error. When you write the patch file, I suggest using several digits (12-14 digits), which should be sufficient to avoid the error message. Perhaps as a long-term solution, we could be less strict, by internally rescaling the patch areas when the area is not exactly 1 (potentially issuing a warning instead of an error).

The CSS file should include living plants only. You are right, you can provide each individual tree as its own cohort in the initialisation file, and FATES will merge them into less cohorts if needed.

**Reply by LianGong98** [reply]
Really appreciate you taking the time to answer my question. It helps me a lot. Thank you！

**Reply by LianGong98** [reply]
Hello, @mpaiao, sorry to bother you again! I have another question. In the parameter file, are **_fates_history_sizeclass_bin_edges_ and _fates_history_height_bin_edges**_ customized based on the **DBH and height** from the inventory data? Are there any defined rules for their definition? Also, is it usually sufficient to keep _**fates_history_ageclass_bin_edges**_ as default, or does it also require customization?

Any assistance would be greatly appreciated!

**Reply by ckoven** [reply]
Hi @LianGong98 the variables `fates_history_sizeclass_bin_edges`, `fates_history_height_bin_edges`, `fates_history_ageclass_bin_edges` in the parameter file are just for how the size-, height-, or age-resolved history output variables are reported, they aren't related to the inventory initialization and won't have any impact on the simulation itself, only on how some of the variables are reported.

**Reply by LianGong98** [reply]
@ckoven Oh, I understand now. Thank you for your response, I really appreciate it!

**Reply by LianGong98** [reply]
Hello @mpaiao  
 I'm having some trouble using the E3SM-FATES-Hydro model to simulate drought deciduous trees Specifically, I have noticed that the leaves do not fall off during the dry season as expected. Additionally, the timing of leaf fall each year seems to be one month later than the previous year's month, for example, if the leaf fall occurs in October in the first year, it happens in November in the second year, and so on.

  I have attempted to adjust some parameters related to leaf phenology, such as fates_phen_drought_threshold and parameters controlling Btran(fates_hydro_avuln_gs ,fates_hydro_p50_gs), but the results are still not satisfactory.
  
  Do you have any insights intowhy the leaves are not shedding during the dry season? Also, why is the timing of leaf fall one month later each year? What other adjustments or optimizations can I try to resolve this issue?
  
I'd really appreciate any assistance you can provide!

Best regards

**Reply by mpaiao** [reply]
@LianGong98 This could be a symptom that the soils are never getting as dry as the value set by `fates_phen_drought_threshold`. The leaf phenology has a first guess for time of year to abscise leaves, and if it never gets sufficiently dry, the leaves will linger for as long as the `fates_turnover_leaf` is set 1 month, and plant will remain leafless for a small period of time (maybe the default is 1 month). So if the conditions are never too dry, this may explain the monthly shift. 

The default value for `fates_phen_drought_threshold=-152957.4` means that the drought-induced abscission will happen when the average soil matric potential across the rooting zone falls below -1.5 MPa. If the soils are sandy, the soil hydraulic conductivity may become very small at relatively low absolute value of matric potential. Maybe try to set the threshold to much less negative values just for testing purposes (e.g., -0.5 MPa), or make the rooting depth of drought deciduous PFTs much shallower than evergreens. Parameters controlling Btran would influence phenology only if the plants close their stomata well before the soil reaches the drought threshold — something I think deciduous plants may not do in nature because they may just abscise leaves before hydraulic failure risk sets in.

**Reply by LianGong98** [reply]
@mpaiao Thank you very much! If I want to make the rooting depth of drought-deciduous trees shallower than that of evergreen trees, should I decrease **fates_allom_fnrt_prof_a** and increase **fates_allom_fnrt_prof_b** or adjust other parameters that control root depth?

**Reply by mpaiao** [reply]
@LianGong98 Yes, these two parameters control the rooting depth. The third one is `fates_allom_fnrt_prof_mode`, which allows changing the functional form of the rooting depth profile, just mind that the meaning of `fates_allom_fnrt_prof_a` and `fates_allom_fnrt_prof_b` may change a bit depending on which functional form you use.

**Reply by LianGong98** [reply]
@mpaiao Thank you for your response. It helped me a lot! I have another question. I'm concerned that the initial conditions might affect the simulation results. I didn't provide any initialization for soil moisture, which could impact the outcomes. How is soil moisture initialization handled in FATES?
